src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -47,6 +47,10 @@ public function __invoke(Node $node, CompilerContext $context)
             return $this->compileBinaryOperator($node, $context);
         }
 
+        if ($node instanceof Node\Scalar\MagicConst\Dir) {
+            return '__DIR__';
```

Comment:
I'd provide a test case where the real value of `__DIR__` compiled by the `NodeCompiler` matches the test suite class's path